lms/static/js/bookmarks/views/bookmarks_list.js

```diff
@@ -59,8 +59,24 @@
                 });
             },
 
-            visitBookmark: function (event) {
-                window.location = event.target.pathname;
+            visitBookmark: function (url) {
+                window.location.href = url;
+            },
+
+            logBookmarkedItem: function (event) {
+                var self = this;
+                var bookmark_id = event.currentTarget.dataset.id;
```

Comment:
Can you look into the jQuery method to read the data attribute? I may be wrong but it may have better browser compatibility.